Uds 2036 - update multiple card render for unity-react-core cards

davidornelas11 opened the pull request:
This adds the options for rendering multiple cards for the ranking cards and Card component in unity-react-core. Webspark will have to be updated to use this but the default/old behavior is still there and users will have to intentinally use the new prop to use the new render feature
### Description

<!-- Description of problem -->
<!-- Solution -->
<!-- Testing Steps -->

## Checklist

- [x] Tests pass for relevant code changes

## Important Reminders
<!-- Add meaningful tests -->
<!-- Remove tests that do not provide value -->

### Links

- [JIRA ticket](https://asudev.jira.com/browse/UDS-2036)

davidornelas11 committed:
feat(unity-react-core): add multiple cards rendering option

davidornelas11 committed:
chore(unity-react-core): fix multiple card tests

asu-jenkins-devops commented:
Storybook deployed at https://unity-uds-staging.s3.us-west-2.amazonaws.com/pr-1577/index.html

davidornelas11 committed:
fix(unity-react-core): update card arrangement render classes

mlsamuelson reviewed (changes requested):
@davidornelas11 I like this approach, it moves us forward without introducing a breaking change. Nice. Appears it will be adaptive for use with multiple column types based on where it's placed.

Only concern at this point is problematic rendering when images have different heights: https://unity-uds-staging.s3.us-west-2.amazonaws.com/pr-1577/@asu/unity-react-core/index.html?path=/story/components-image--three-images-arrangement 

<img width="1473" height="676" alt="Screenshot 2025-09-12 at 2 24 53 PM" src="https://github.com/user-attachments/assets/a5097847-b8fa-4999-aea4-f16e49a17d07" />

davidornelas11 committed:
chore(unity-react-core): update card stories for grid examples

davidornelas11 committed:
fix(unity-bootstrap-theme): fix card drop shadow in multiple cards arrangement

davidornelas11 committed:
fix(unity-react-core): update vertical layout options

davidornelas11 committed:
fix(unity-bootstrap-theme): update arrangement styles for image component

The aspect ratio of 4:3 is enforced when the .uds-card-arrangement is wrapped around an image
component. No changes to previous behavior

davidornelas11 commented:
Per brand team, a 4:3 aspect ratio will be enforced with the styles when the card-arrangement is used. No changes to previous behavior. Webspark already crops and takes care of this so will only affect nonwebspark users

scott-williams-az commented on packages/unity-react-core/src/components/Card/Card.jsx:
```diff
@@ -152,6 +214,57 @@ Card.propTypes = {
+  /**
+   * Array of card objects for rendering multiple cards
+   */
+  cards: PropTypes.arrayOf(
```
Cards has same structure as BaseCard, might need to rearrange the file, but can we reuse the prop structure

cards: PropTypes.arrayOf(PropTypes.shape(BaseCard.propTypes))

scott-williams-az commented on packages/unity-react-core/src/components/Card/Card.jsx:
```diff
@@ -46,7 +46,69 @@ export const Card = ({
   showBorders = true,
   cardLink,
+  cards = [],
+  columns = "0",
```
I think we are moving towards responsive design, and we should not allow columns to be a property. Columns should be responsive and allow the card to adjust between the min and max widths provided by the brand team.

scott-williams-az commented on packages/unity-react-core/src/components/Card/Card.jsx:
```diff
@@ -46,7 +46,69 @@ export const Card = ({
   tags,
   showBorders = true,
   cardLink,
+  cards = [],
```
Card feels like it is doing too much. Should we have a card-arrangement component that takes an array of 1 or more cards?

scott-williams-az replied:
I am ok with this temporarily if we have to keep it for WS2 purposes, but think it will cause issue down the line since it is a strange pattern

scott-williams-az commented on packages/unity-react-core/src/components/Card/Card.jsx:
```diff
@@ -46,7 +46,69 @@ export const Card = ({
   cardLink,
+  cards = [],
+  columns = "0",
+  layout = "auto"
```
Does `auto` mean horizontal?
Instead of `layout` default to `auto` can we change `layout` prop to `vertical` = `true` or `forceVertical` = `true`?

scott-williams-az replied:
I see we already have  horizontal prop. these props are getting confusing

davidornelas11 commented:
Closing in favor of https://github.com/ASU/asu-unity-stack/pull/1603
